crypto/rsa/rsa_ossl.c

```diff
@@ -578,7 +580,10 @@ static int rsa_ossl_public_decrypt(int flen, const unsigned char *from,
         r = RSA_padding_check_X931(to, num, buf, i, num);
         break;
     case RSA_NO_PADDING:
-        memcpy(to, buf, (r = i));
+        if (i > 0)
+            memcpy(to, buf, (r = i));
```

Comment:
is there something wrong with memcpy(to, buf, 0) ?

Reply:
would the same work here?
```C
if (i > 0)
    memcpy(to, buf, (r = i));
break;
```

Reply:
If ```to``` or ```buf``` is NULL or otherwise invalid, calling memcpy with size 0 is undefined.
https://stackoverflow.com/questions/5243012/is-it-guaranteed-to-be-safe-to-perform-memcpy0-0-0